### PR TITLE
Update installation instructions

### PR DESCRIPTION
**Pull request by feedmeapples**
## What was changed

Added CDN links in installation instructions

## Why?

CDN instructions were not clear

## Checklist

1. Closes 

2. How was this tested:

3. Any docs updates needed?

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -33,10 +33,15 @@ Download and extract the latest release from [GitHub releases](https://github.co
...
-4. Add the `temporal` binary to your system PATH.
+  1. Select the platform and architecture needed.
+     - <a href="https://temporal.download/cli/archive/latest?platform=linux&arch=amd64">Download for Linux amd64</a>
+     - <a href="https://temporal.download/cli/archive/latest?platform=linux&arch=arm64">Download for Linux arm64</a>
+     - <a href="https://temporal.download/cli/archive/latest?platform=darwin&arch=amd64">Download for macOS amd64</a>
+     - <a href="https://temporal.download/cli/archive/latest?platform=darwin&arch=arm64">Download for macOS arm64</a>
+     - <a href="https://temporal.download/cli/archive/latest?platform=windows&arch=amd64">Download for Windows amd64</a>
+     - <a href="https://temporal.download/cli/archive/latest?platform=windows&arch=arm64">Download for Windows arm64</a>
+  2. Extract the downloaded archive.
+  3. Add the `temporal` binary to your PATH.
```
Comment (feedmeapples) [inline README.md:44]:
```suggestion
  3. Add the `temporal` binary to your PATH. `temporal.exe` for Windows
```

**Review by lorensr — APPROVED** [review]

File: README.md [hunk 2]
```diff
@@ -33,10 +33,15 @@ Download and extract the latest release from [GitHub releases](https://github.co
...
 
-1. Select the binary with the corresponding platform and architecture of your system.
-2. Download the binary.
-3. Extract the downloaded archive.
-4. Add the `temporal` binary to your system PATH.
+  1. Download:
+     - <a href="https://temporal.download/cli/archive/latest?platform=linux&arch=amd64">Download for Linux amd64</a>
+     - <a href="https://temporal.download/cli/archive/latest?platform=linux&arch=arm64">Download for Linux arm64</a>
+     - <a href="https://temporal.download/cli/archive/latest?platform=darwin&arch=amd64">Download for macOS amd64</a>
+     - <a href="https://temporal.download/cli/archive/latest?platform=darwin&arch=arm64">Download for macOS arm64</a> (Apple silicon)
```
Comment (feedmeapples) [inline README.md:40]:
```suggestion
     - <a href="https://temporal.download/cli/archive/latest?platform=darwin&arch=arm64">Download for macOS arm64 (Apple silicon) </a>
```

**Comment by bergundy** [comment]
@lorensr, @DJSanti, @rachfop can you guys go over this README, https://docs.temporal.io/cli/, and https://docs.temporal.io/application-development/foundations#run-a-development-server and borrow the best parts from each doc?

**Comment by lorensr** [comment]
@bergundy borrow to improve this README? sure, started in #172 and can add from the two linked places

**Pull request merged**